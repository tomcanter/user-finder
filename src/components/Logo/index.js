// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';

export default function Logo() {
  return (
    <div>
      <h1 className={css(styles.Logo_title)}>GitHub User Search</h1>
      <p className={css(styles.Logo_subTitle)}>
        Browse users and their profiles via <a href="https://developer.github.com/v3/">the GitHub API</a>
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  Logo_title: {
    fontSize: 28,
  },

  Logo_subTitle: {
    marginTop: 5,
    fontSize: 14,
  },
});
