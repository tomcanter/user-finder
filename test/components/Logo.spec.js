import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import {
  BrowserRouter as Router,
} from 'react-router-dom';

import Logo from 'components/Logo';

describe('Component: Logo', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Router>
        <Logo />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
