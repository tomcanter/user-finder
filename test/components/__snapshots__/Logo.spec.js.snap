// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Logo should render 1`] = `
<div>
  <h1
    className="Logo_title_5kripx"
  >
    <a
      className="Logo_link_18deiwy"
      href="/"
      onClick={[Function]}
    >
      GitHub User Search
    </a>
  </h1>
  <p
    className="Logo_subTitle_1r70osc"
  >
    Browse users and their profiles via 
    <a
      href="https://developer.github.com/v3/"
    >
      the GitHub API
    </a>
  </p>
</div>
`;
